File: livre/src/main/java/fr/deroffal/bibliotheque/livre/adapter/controller/AuthentificationController.java
package fr.deroffal.bibliotheque.livre.adapter.controller;

import static org.springframework.http.HttpStatus.OK;

import fr.deroffal.bibliotheque.securite.auth.AuthentificationService;
import fr.deroffal.bibliotheque.securite.auth.JwtRequest;
import fr.deroffal.bibliotheque.securite.auth.JwtResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

@RestController
@CrossOrigin
@RequiredArgsConstructor
public class AuthentificationController {

    private final AuthentificationService authentificationService;

    @PostMapping("/authenticate")
    @ResponseStatus(OK)
    public JwtResponse authenticate(@RequestBody final JwtRequest authenticationRequest) {
        return authentificationService.authenticate(authenticationRequest);
    }
}

File: livre/src/main/kotlin/fr/deroffal/bibliotheque/livre/adapter/controller/AuthentificationController.kt
package fr.deroffal.bibliotheque.livre.adapter.controller

import fr.deroffal.bibliotheque.securite.auth.AuthentificationService
import fr.deroffal.bibliotheque.securite.auth.JwtRequest
import fr.deroffal.bibliotheque.securite.auth.JwtResponse
import lombok.RequiredArgsConstructor
import org.springframework.http.HttpStatus.OK
import org.springframework.web.bind.annotation.*

@RestController
@CrossOrigin
@RequiredArgsConstructor
class AuthentificationController(private val authentificationService: AuthentificationService) {

    @PostMapping("/authenticate")
    @ResponseStatus(OK)
    fun authenticate(@RequestBody authenticationRequest: JwtRequest): JwtResponse {
        return authentificationService.authenticate(authenticationRequest)
    }
}
